src/react/GroupDevicesPage.js

```diff
@@ -72,12 +85,46 @@ export default class GroupDevicesPage extends React.Component {
     });
   }
 
+  showModal(device) {
+    this.setState({
+      showModal: true,
+      selectedDeviceName: device.name,
+      selectedDeviceId: device.id
+    });
+  }
+
+  handleModalCancel() {
+    this.setState({
+      showModal: false
+    });
+  }
+
+  removeDevice() {
+    this.setState({
+      deletingDevice: true
```

Comment:
This should probably be "removingDevice" for consistency with the operation name